src/middlewares/checkAuth.js

```diff
@@ -1,20 +1,24 @@
 import jwt from 'jsonwebtoken';
+import localStorage from 'localStorage';
 
 const auth = (req, res, next) => {
-  const token = !req.headers.token ? req.params.token : req.headers.token;
+  const token = localStorage.getItem('token');
   if (!token) {
-    return res.status(401).json({ error: 'Access denied. no token provided.' });
+    return res.status(401).json({
+      status: 401,
+      error: 'you are not logged in'
+    });
   }
   try {
-    const decoded = jwt.verify(token, process.env.JWT);
+    const decoded = jwt.verify(token, process.env.SECRETKEY);
     req.user = decoded;
```

Comment:
@Niyonsengaeric  this value req.user, if it wasn't used anywhere should be removed or use it in the controller you are protecting as it helps you to keep the decoded data you have saved in token.

Reply:
the reason for this req.user is to store the decoded information so that it can be used in different controllers